### ISSUE TITLE
linter: unicorn/prefer-event-target allow EventEmitter from packages

### ISSUE THREAD

**Issue by tlebreton** [oxc-project/oxc#17371]
Hi,

For the rule unicorn/prefer-event-target in the original plugin they made a modification to allow EventEmitter from certain packages and in particular from `@angular/core` .

https://github.com/sindresorhus/eslint-plugin-unicorn/issues/1853

Is it possible to check the origin of the EventEmitter used to see if it's from global import or from some packages and allow EventEmitter from packages ?

**Comment by connorshea** [comment]
I don't see a strong reason not to make this same change. 

ESLint plugin Docs for reference: https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/prefer-event-target.md

Code change that was made upstream: https://github.com/sindresorhus/eslint-plugin-unicorn/pull/2197/files

Current rule source code from original rule: https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/rules/prefer-event-target.js

Current tests from original rule: https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/test/prefer-event-target.js

The following test cases should start passing after this change is made, they should be added to the tests for the prefer-event-target oxlint rule:

```js
import { EventEmitter } from "@angular/core";
class Foo extends EventEmitter {}
```

```js
const { EventEmitter } = require("@angular/core");
class Foo extends EventEmitter {}
```
```js
const EventEmitter = require("@angular/core").EventEmitter;
class Foo extends EventEmitter {}
```
```js
import { EventEmitter } from "eventemitter3";
class Foo extends EventEmitter {}
```
```js
const { EventEmitter } = await import("eventemitter3");
class Foo extends EventEmitter {}
```
```js
const EventEmitter = (await import("eventemitter3")).EventEmitter;
class Foo extends EventEmitter {}
```

**Comment by tlebreton** [comment]
Hi,

Is there any reason why the PR linked to this issue has been closed @connorshea  ?
